### PR TITLE
runecraft: give minimap icons their own overlay

### PR DESCRIPTION
**Pull request by Hydrox6**
Moving the AbyssOverlay down to the correct layer stopped the runes from displaying on the minimap. Why an overlay was trying to do both Scene and Minimmap rendering I have no idea, but I've moved the minimap code to its own overlay. I've also moved the "what rifts to render" stuff to the plugin, since it's now common

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/AbyssOverlay.java [hunk 1]
```diff
@@ -127,7 +114,7 @@ private void highlightDarkMage(Graphics2D graphics)
 	private void renderRifts(Graphics2D graphics, DecorativeObject object)
 	{
 		AbyssRifts rift = AbyssRifts.getRift(object.getId());
-		if (rift == null || !rifts.contains(rift))
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/AbyssOverlay.java]: Then this field needs to be removed

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/AbyssMinimapOverlay.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+import java.util.HashMap;
+import java.util.Map;
+
+public class AbyssMinimapOverlay extends Overlay
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/AbyssMinimapOverlay.java]: `class AbyssMinimapOverlay extends Overlay`

File: runelite-client/src/main/java/net/runelite/client/plugins/runecraft/AbyssMinimapOverlay.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+			if (miniMapImage != null)
+			{
+				graphics.drawImage(image, miniMapImage.getX(), miniMapImage.getY(), null);
+			}
+		}
+
+		return null;
+	}
+
+	public BufferedImage getImage(AbyssRifts rift)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/runecraft/AbyssMinimapOverlay.java]:
```suggestion
	private BufferedImage getImage(AbyssRifts rift)
```

Comment (Hydrox6) [reply runelite-client/src/main/java/net/runelite/client/plugins/runecraft/AbyssMinimapOverlay.java]: Turns out it was public in the original overlay, weirdly. Either way, fixed

**Pull request merged**